xbmc/utils/SeekHandler.cpp

```diff
@@ -79,16 +160,45 @@ bool CSeekHandler::InProgress() const
 
 void CSeekHandler::Process()
 {
-  if (m_timer.GetElapsedMilliseconds() > time_before_seek)
+  if (m_timer.GetElapsedMilliseconds() > m_seekDelay)
   {
     if (!g_infoManager.m_performingSeek && m_timer.GetElapsedMilliseconds() > time_for_display) // TODO: Why?
       g_infoManager.SetSeeking(false);
+
     if (m_requireSeek)
     {
       g_infoManager.m_performingSeek = true;
+
+      // calculate the seek time
       double time = g_infoManager.GetTotalPlayTime() * m_percent * 0.01;
```

Comment:
As discussed at IRC what you've said here will not fix the issue with the 1sec offset issue and
is not related to the SeekHandler calculation.